src/core/auth/qgsauthmanager.cpp

```diff
@@ -83,15 +82,29 @@ static const QString sDescription = QObject::tr( "Master Password <-> KeyChain s
 #endif
 
 
+
 QgsAuthManager *QgsAuthManager::instance()
 {
   if ( !sInstance )
   {
-    sInstance = new QgsAuthManager();
+    static QMutex sMutex;
+    QMutexLocker locker( &sMutex );
+    if ( !sInstance )
```

Comment:
They are different: this is static and the other one is bound to the instance, but, to be honest I don't think that  it is strictly necessary, since only QgsApplication can call this function and there is no chance that it can be called from multiple threads.
I put it in just because it was the established pattern used by other refactored singletons see for example 
https://github.com/qgis/QGIS/commit/e59f1d5d4840dedb750b7028ccc086b7e05519da and since I don't see any difference in the way those two singleton are used (they are instanciated at the same time from the same function) I thought that if that change was accepted it means that it was the right thing to do even if I didn't understand the necessity of it, furthermore that mutex does not hurt, of that I'm sure.
Did I miss something?
Btw, I'll remove it because I also don't see a valid reason for that mutex but I'd like to know what is the reason for having the static mutex in https://github.com/qgis/QGIS/commit/e59f1d5d4840dedb750b7028ccc086b7e05519da @nyalldawson any comment?

Reply:
But yes, we need a second conditional because the mutex is locked after the first conditional and a race condition could happen in between.